src/lib.rs

```diff
@@ -679,6 +680,74 @@ pub fn valid_header_value<'a>(headers: &HeaderMap, header: (&'a str, &'a str)) -
     }
 }
 
+/// Helper to confirm the URI is valid and local.
+fn valid_local_uri(user: &mut GooseUser, uri: &str) -> bool {
+    match uri.parse::<Uri>() {
+        Ok(parsed_uri) => {
+            if let Some(parsed_host) = parsed_uri.host() {
+                if parsed_host == user.base_url.host_str().unwrap() {
+                    // The URI host matches the base_url.
```

Comment:
My only note here is that I suspect this logic may result in a change in behavior when the `base_url` includes a path. I don't know if that is a supported usage necessarily, but [these tests](https://github.com/tag1consulting/goose/blob/58e9da3fcce200084e1107248bb6783b6528822c/src/goose.rs#L3227-L3229) look like it may be?

Consider a case where `base_url` is something like `http://example.com/my/base/` and you have two assets:
* A: `http://example.com/foo.js`
* B: `http://example.com/my/base/bar.js`

I think the [prior behavior](https://github.com/tag1consulting/goose-eggs/blob/827f5ce6489542fd1536d5b3b3276394595187b9/src/lib.rs#L736) would have matched case B but not case A because it included the `base_url`'s path in the regexp.

However, the new behavior here would match both case A and case B because the `Host` portion matches and the `base_url`'s path is not checked.

Personally, I think this is reasonable and don't consider it an issue / don't want to complicate things further, just highlighting this in case there could be an expectation that only resources under the base URL's path are loaded. :)

Disclaimer: I have only visually inspected this and not tested or verified this thought.

Reply:
I'm mixed as if this should be fixed and/or is a bug. I could see this being a problem in a multi-site installation, where multiple sites are served from different sub-directories. But in this case, any common files included from the base installation etc likely should be included in a load test. Is there another use-case where this may happen, where it would generally be considered invalid to include the files?

I'm going to merge as-is for now, and we can revisit this if it proves problematic.

Reply:
I don't think this is a problem and agree with your assessment, just noting that it's a change. :)